ibis/expr/operations/generic.py

```diff
@@ -257,6 +288,9 @@ class Literal(ValueOp):
     )
     dtype = Arg(rlz.datatype)
 
+    def _validate(self):
+        self.value = _normalize(self.dtype, self.value)
```

Comment:
Currently hooking in here since `_validate` already exists, but happy to add something purpose-built

Reply:
I think you can call `_normalize(value)` here instead: https://github.com/ibis-project/ibis/blob/master/ibis/expr/types.py#L1242

Reply:
How about adding the functionality to `datatypes.py` instead?

Reply:
Moved `_normalize` into `types.py` and defined the `Dispatcher` in `datatypes.py`